sky/serve/autoscalers.py

```diff
@@ -16,32 +17,37 @@
 
 logger = sky_logging.init_logger(__name__)
 
-_UPSCALE_DELAY_S = 300
-_DOWNSCALE_DELAY_S = 1200
-
 
 class AutoscalerDecisionOperator(enum.Enum):
     SCALE_UP = 'scale_up'
     SCALE_DOWN = 'scale_down'
-    NO_OP = 'no_op'
 
 
 @dataclasses.dataclass
 class AutoscalerDecision:
     """Autoscaling decisions.
 
-    |---------------------------------------------------------|
-    | Operator   | TargetType | Meaning                       |
-    |------------|------------|-------------------------------|
-    | SCALE_UP   | int        | Number of replicas to add     |
-    |------------|------------|-------------------------------|
-    | SCALE_DOWN | List[int]  | List of replica ids to remove |
-    |------------|------------|-------------------------------|
-    | NO_OP      | None       | No scaling needed             |
-    |---------------------------------------------------------|
+    |------------------------------------------------------------------------|
+    | Operator   | TargetType                | Meaning                       |
+    |------------|---------------------------|-------------------------------|
+    | SCALE_UP   | Optional[Dict[str, Any]   | Resource override to add      |
+    |------------|---------------------------|-------------------------------|
+    | SCALE_DOWN | int                       | Replica id to remove          |
+    |------------------------------------------------------------------------|
     """
     operator: AutoscalerDecisionOperator
-    target: Optional[Union[int, List[int]]]
+    target: Union[Optional[Dict[str, Any]], int]
+
+    # TODO(MaoZiming): Add a doc to elaborate on autoscaling policies.
+    def __init__(self,
+                 operator: AutoscalerDecisionOperator,
+                 target: Union[Optional[Dict[str, Any]], int] = None):
```

Comment:
```suggestion
                 target: Union[Optional[Dict[str, Any]], int]):
```
We don't need a default argument here? Reason: we should explicitly specify autoscaling target in every construction of `AutoscalerDecision`.